scala/scala_2.13/src/main/scala/com/linecorp/armeria/scala/package.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria
+
+import com.google.common.util.concurrent.MoreExecutors
+import com.linecorp.armeria.common.annotation.UnstableApi
+
+import _root_.scala.concurrent.ExecutionContext
+
+/**
+ * Provides a collection of utilities for using Armeria in a project written in Scala.
+ * Read [[https://armeria.dev/docs/advanced-scala]] for more information.
+ */
+package object scala {
+
+  /**
+   * An `ExecutionContext` that runs the submitted `Runnable`s immediately on the thread that calls `execute()`
+   * and then yields back control to the caller after the `Runnable` finishes its execution.
+   *
+   * This is often useful when you need to avoid unnecessary context switches for the short-running tasks or
+   * asynchronous callbacks that are OK to run from an event loop thread.
+   *
+   * You must *not* use this `ExecutionContext` for a potentially long-running tasks.
+   */
+  @UnstableApi
+  val directExecutionContext: ExecutionContext = {
```

Comment:
This has traditionally been called `SameThreadExecutionContext` and 2.13 calls it `parasitic` ( I hadn't actually seen that before reviewing this PR). Does it make sense to introduce yet another name for this concept or could you use one of the existing ones?

Reply:
I initially tried to use `ExecutionContext.parasitic` but I gave up because:

- 2.12 doesn't have it; and
- It uses `BatchExecutor` which has some overhead in our case.

Should we rename it to `parasitic` or `sameThreadExecutionContext`?